ngx-fudis/projects/ngx-fudis/src/lib/components/pagination/pagination.component.ts

```diff
@@ -0,0 +1,342 @@
+
+  /**
+   * Next button aria-label from Fudis translation keys
+   */
+  protected _paginationNextButtonAria = new BehaviorSubject<string>('');
+
+  /**
+   * Main CSS classes for buttons
+   */
+  protected _mainCssPrevClass: BehaviorSubject<string> = new BehaviorSubject<string>(
+    'fudis-pagination-button__previous',
+  );
+  protected _mainCssNextClass: BehaviorSubject<string> = new BehaviorSubject<string>(
+    'fudis-pagination-button__next',
+  );
+
+  /**
+   * Returns an array of numbers from given starting to ending number
+   */
+  protected range = (start: number, end: number): number[] => {
+    const length = end - start + 1;
+    return Array.from({ length }, (_, i) => start + i);
+  };
+
+  /**
+   * HTML id
+   */
+  protected _id: string;
+
+  /**
+   * Private signal for counting Pagination item visible siblings
+   */
+  private _siblingCount = signal(2);
+
+  /**
+   * After DOM is updated, restore focus to the new active element
+   */
+  private userSelectedIndex = -1;
+
+  /**
+   * Stores the previous page index for change detection
+   */
+  private prevPageIndex: number | undefined;
+
+  /**
+   * To observe container size changes and adjust Pagination sibling count based on viewport changes
+   */
+  private observer?: ResizeObserver;
+
+  /**
+   * Computed list of pagination items (page numbers + ellipses)\
+   * Recalculated whenever page count, current page, or sibling count changes.
+   */
+  itemList = computed(() =>
+    this.createPaginationItemList(this._pageCount(), this._pageIndex(), this._siblingCount()),
+  );
+
+  /**
+   * Get translation for aria-live page openend announcement
+   */
+  get liveAnnouncement(): string {
+    if (this._pageCount() === 0) return '';
+    return (
+      this._translationService.getTranslations()().PAGINATION.OPENED_PAGE +
+      ` ${this._pageIndex() + 1}`
+    );
+  }
+
+  /**
+   * Set button collapsed styles if Pagination container is smaller than < 465px
+   */
+  private _setClasses(collapsed: boolean): void {
+    if (collapsed) {
+      this._mainCssPrevClass.next('fudis-pagination-button__previous--collapsed');
+      this._mainCssNextClass.next('fudis-pagination-button__next--collapsed');
+    } else {
+      this._mainCssPrevClass.next('fudis-pagination-button__previous');
+      this._mainCssNextClass.next('fudis-pagination-button__next');
+    }
+  }
+
+  /**
+   * Create function that determines when to show ellipsis on start or end of the pagination list
+   */
+  private createPaginationItemList(
+    pageCount: number,
+    pageIndex: number,
+    siblingCount: number,
+  ): (Ellipsis | number)[] {
+    const startPages = this.range(1, Math.min(1, pageCount));
+    const endPages = this.range(Math.max(pageCount, 2), pageCount);
+
+    /**
+     * Calculate the first sibling page number to display around the current page
+     */
+    const siblingsStart = Math.max(
+      Math.min(pageIndex + 1 - siblingCount, pageCount - 1 - siblingCount * 2 - 1),
```

Comment:
This might be personal but this is **very** hard to read and I'm not sure (without testing) does the calculation work "properly" without parentheses.

Is the calculation pattern `(pageIndex + 1) - siblingCount, ((pageCount - 1) - (siblingCount * 2)) - 1 `or something else? So could this and the rest of the similar calculations have bit more structure with the parentheses for example.